astropy/io/ascii/core.py

```diff
@@ -1466,7 +1472,13 @@ def _get_writer(Writer, fast_writer, **kwargs):
     if 'exclude_names' in kwargs:
         writer.exclude_names = kwargs['exclude_names']
     if 'fill_values' in kwargs:
-        writer.data.fill_values = kwargs['fill_values']
+        # Prepend user-specified values to the class default.
+        with suppress(TypeError, IndexError):
+            # Test if it looks like (match, replace_string, optional_colname),
+            # in which case make it a list
+            kwargs['fill_values'][1] + ''
+            kwargs['fill_values'] = [kwargs['fill_values']]
+        writer.data.fill_values = kwargs['fill_values'] + writer.data.fill_values
```

Comment:
@hamogu - I'm getting a little confused about whether you talking about reading or writing, since you mentioned getting files from your collaborators and replacing -999 with missing.  For writing you would have said "replacing missing with -999".

It's important to note that the user-supplied kwarg will always override the default by virtue of being earlier in the list.  I also think there will be no change in behavior for custom writer classes that define a custom `fill_value`, e.g. to output -999 for missing values.